### PR TITLE
Improve Curve3D debug drawing

### LINKED ISSUE #83634: Curved Path3D not showing as curved when using "Visible Paths" Debug Option
### Godot version

v4.1.2.stable.flathub [399c9dc39]

### System information

Godot v4.1.2.stable (399c9dc39) - Freedesktop SDK 23.08 (Flatpak runtime) - Wayland - Vulkan (Compatibility) - NVIDIA GeForce GTX 1050 Ti with Max-Q Design () - Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 Threads)

### Issue description

When creating curved Path3Ds and using the Option `Debug > Visible Paths` they don't show up curved and instead are just straight lines.
Source Code: https://github.com/Arbee4ever/Train-Game/blob/master/scripts/World.gd#L115

### Steps to reproduce

1. Create Script to create curved Path3D
2. enable `Debug > Visible Paths`
3. run game
4. see straight line instead of curve

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by BlueCube3310**
Fixes #83634

Changes the way 3d debugging curves are drawn at runtime to be more in line with how they are drawn in the editor.

Comparison:
curve_pr

### CODE REVIEW

File: scene/3d/path_3d.cpp [hunk 1]
```diff
@@ -93,24 +93,60 @@ void Path3D::_update_debug_mesh() {
...
+		// Path3D as a ribbon.
+		ribbon.write[i] = p1;
+
+		// Fish Bone.
+		const Vector3 p_left = p1 + (side + forward - up * 0.3) * 0.06;
+		const Vector3 p_right = p1 + (-side + forward - up * 0.3) * 0.06;
+
+		const int bone_idx = i * 4;
+
+		bones.write[bone_idx] = p1;
```
Comment (AThousandShips) [inline scene/3d/path_3d.cpp:129]:
```suggestion
		bones_ptr[bone_idx] = p1;
```

File: scene/3d/path_3d.cpp [hunk 2]
```diff
@@ -93,24 +93,60 @@ void Path3D::_update_debug_mesh() {
...
+		bones.write[bone_idx + 1] = p_left;
```
Comment (AThousandShips) [inline scene/3d/path_3d.cpp:130]:
```suggestion
		bones_ptr[bone_idx + 1] = p_left;
```

File: scene/3d/path_3d.cpp [hunk 3]
```diff
@@ -93,24 +93,60 @@ void Path3D::_update_debug_mesh() {
...
+		bones.write[bone_idx + 2] = p1;
```
Comment (AThousandShips) [inline scene/3d/path_3d.cpp:131]:
```suggestion
		bones_ptr[bone_idx + 2] = p1;
```

File: scene/3d/path_3d.cpp [hunk 4]
```diff
@@ -93,24 +93,60 @@ void Path3D::_update_debug_mesh() {
...
+		bones.write[bone_idx + 3] = p_right;
```
Comment (AThousandShips) [inline scene/3d/path_3d.cpp:132]:
```suggestion
		bones_ptr[bone_idx + 3] = p_right;
```

File: scene/3d/path_3d.cpp [hunk 5]
```diff
@@ -93,24 +93,63 @@ void Path3D::_update_debug_mesh() {
...
+
+		const Vector3 p1 = r.origin;
+		const Vector3 side = r.basis.get_column(0);
+		const Vector3 up = r.basis.get_column(1);
+		const Vector3 forward = r.basis.get_column(2);
+
+		// Path3D as a ribbon.
+		ribbon_ptr[i] = p1;
+
+		// Fish Bone.
```
Comment (akien-mga) [inline scene/3d/path_3d.cpp:126]: Very nitpicky, not necessarily worth updating the PR just for this. But if other review comments require more changes, could be done alongside.
```suggestion
		// Fish bone.
```

**Review by Calinou — APPROVED** [review]
Tested locally with https://github.com/Calinou/godot-reflection/tree/add-benchmark (rebased on top of `master` 2f73a059cefadcd944b6874f2557ec82e46a562d), it works as expected. Great work :slightly_smiling_face: 

Before | After *(this PR)*
-|-
Screenshot_20231207_005444 webp | Screenshot_20231207_005432 webp

Fun fact: You can use `--debug-paths` when opening the editor to compare the editor and runtime visualization. This is with `master`:

image

With your PR, they overlap perfectly.

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**